bento/src/main/java/com/yelp/android/bento/core/Component.java

```diff
@@ -298,6 +306,32 @@ public void setSpanSizeLookup(@NonNull SpanSizeLookup lookup) {
         mSpanSizeLookup = lookup;
     }
 
+    /**
+     * Override this method to handle reordering of items.
+     *
+     * @param oldIndex The index the item was originally in.
+     * @param newIndex The index the item was moved to.
+     * @see Component#isReorderable() (boolean)
+     */
+    public void onItemsMoved(int oldIndex, int newIndex) {
+
+    }
+
+    protected RangedValue<Component> getLowestRangeValue(int index) {
+        return new RangedValue<>(this, new Range(0, getCount()));
+    }
+
+    /**
+     * Returns the lowest component in the component tree. In general, a {@link ComponentGroup} will
+     * recursively call this method, removing the lower bound of its range each time. A
+     * {@link Component} should just return itself.
+     * @param index The index to search for.
+     * @return The lowest component (leaf) in the component tree.
+     */
+    public Component getLowestComponentAtIndex(int index) {
```

Comment:
Instead of putting this in the Component class, I think we should do something similar to the `findComponentOffset` in ComponentGroup where we just recursively search the children.

Reply:
Also, what if `index` is not `0 <= index < count`